app/src/main/java/com/example/parkingsystem/activity/ParkingActivity.java

```diff
@@ -23,6 +24,10 @@ protected void onCreate(Bundle savedInstanceState) {
         presenter = new ParkingPresenter(new ParkingModel(), new ParkingView(this, binding));
 
         binding.buttonMainSetAmountParkingSpaces.setOnClickListener(view -> presenter.inflateDialog(this));
+        binding.buttonMainReservation.setOnClickListener(v -> {
+            Intent intent = new Intent(ParkingActivity.this, ReservationActivity.class);
```

Comment:
this is not correct, you are not doing the MVP... the button need to call a method in the presenter, presenter to the view and in the view is correct to do the startActivity

Reply:
Fixed